### PR TITLE
Tests for db functions

### PR DESCRIPTION
**Pull request by irfanbacker**
I have added tests for `PasswordStore` and `FileStore`.

There is an issue regarding the datatypes in `File` structure. Currently, all attributes are non-nullable. But there arises a case when `FileStore.ClearGroup()` method is used when `group_id` becomes `NULL` (The test for this scenario has been commented right now [here](https://github.com/irfanbacker/actual-sync/blob/9f55c850b5abf58875e41629e70c7167a4bfe8a2/internal/storage/sqlite/file_store_test.go#L163)). When this happens, All functions which scans data from the database will return errors since we'll be trying to assign nil to a string type. One way to solve this is to use `sql.Null<TypeName>` types.

What do you suggest we do here @nathanjisaac ?

### CODE REVIEW

**Comment by nathanjisaac** [comment]
@irfanbacker I haven't confirmed this but from [this article](https://medium.com/@luismasuelli/lets-order-a-pizza-in-go-null-fields-32782121d3e7) we would need to use the pointer version of those fields. 

```go
type Foo struct {
   ID    uint // not nullable
   Notes *string // nullable 
}
```

**Comment by nathanjisaac** [comment]
I'm honestly not a huge fan of how go handles nullable types. I would much prefer an optional type like [this library](https://arrow-kt.io/docs/apidocs/arrow-core/arrow.core/-option/) in kotlin. I'm still looking for a good solution but it looks like pointers might be the answer for now. 

https://nilsmagnus.github.io/post/nillability-in-go/

**Comment by nathanjisaac** [comment]
I guess this is kind of what I'm looking for. Not sure I want to use this but it is the idea I am looking for. 

https://github.com/markphelps/optional

**Comment by irfanbacker** [comment]
> I guess this is kind of what I'm looking for. Not sure I want to use this but it is the idea I am looking for. 
> 
> https://github.com/markphelps/optional

Have you looked at the nulltypes in database/sql library? Those are very similar to this. Since we are already using that library, we could use these types. What do you think?

**Comment by nathanjisaac** [comment]
**TL;DR - Try using an empty string instead of nil.** 

I haven't really made up my mind on the best option. The downside I see to using the `sql.NullString` value type is that it's a `sql` specific value type. Since our data layer is suppose to be  data-source agnostic it I have a hard time thinking that using the `sql.NullString` type is a good idea. 

If we choose not to use the `sql.NullString` value type then we have a few other options. 

1. Use the pointers
2. Don't use null values - default to empty string?
3. Create our own `core.NullString` type

I think we could probably get away with using an empty string instead of a null value. I don't love having to check for nil's all over the place. 

I think for now I'd try to stick with an empty string and see if we can get away with that for now. 

---

This is another useful article that I found outlines the options pretty clearly. 

https://iamdual.com/en/posts/handle-sql-null-golang/

**Comment by irfanbacker** [comment]
> I think we could probably get away with using an empty string instead of a null value. I don't love having to check for nil's all over the place. 

This could actually work in this case since `group_id` is a generated uuid when not null.

**Comment by irfanbacker** [comment]
@nathanjisaac I have now added support for NULL `group_id` in the functions. It will be represented as `""` in Go.
